generate_linux_tests.rb

```diff
@@ -0,0 +1,245 @@
+        count += 1
+
+        file.write '            ("' + func_name + '", ' + func_name + (count == class_count ? ")\n" : "),\n")
+      end
+
+      file.write "        ]\n"
+      file.write "    }\n"
+      file.write "}\n"
+    end
+  end
+end
+
+def create_linux_main(tests_directory, all_test_sub_directories, files)
+  file_name = tests_directory + '/LinuxMain.swift'
+  print 'Creating file: ' + file_name + "\n"
+
+  File.open(file_name, 'w') do |file|
+    file.write header(file_name)
+    file.write "\n"
+
+    file.write "#if os(Linux) || os(FreeBSD)\n"
+    for test_sub_directory in all_test_sub_directories.sort { |x, y| x <=> y }
+      file.write '    @testable import ' + test_sub_directory + "\n"
+    end
+    file.write "\n"
+    file.write "    XCTMain(\n        [\n"
+
+    test_cases = []
+    for classes in files
+      for class_array in classes
+        test_cases << class_array[0]
+      end
+    end
+
+    cases_count = test_cases.size
+    count = 0
+    for test_case in test_cases.sort { |x, y| x <=> y }
+      count += 1
+
+      file.write '            testCase(' + test_case + (count == cases_count ? ".allTests)\n" : ".allTests),\n")
+    end
+    file.write "        ]\n    )\n"
+    file.write "#endif\n"
+  end
+end
+
+def parse_source_file(file_name)
+  puts 'Parsing file: ' + file_name + "\n"
+
+  classes = []
+  current_class = nil
+  in_if_linux = false
+  in_else = false
+  ignore = false
+
+  #
+  # Read the file line by line
+  # and parse to find the class
+  # names and func names
+  #
+  File.readlines(file_name).each do |line|
+    if in_if_linux
+      if /\#else/ =~ line
+        in_else = true
+        ignore = true
+      elsif /\#end/ =~ line
+          in_else = false
+          in_if_linux = false
+          ignore = false
+      end
+    elsif /\#if[ \t]+os\(Linux\)/ =~ line
+        in_if_linux = true
+        ignore = false
+    end
+
+    next if ignore
+    # Match class or func
+    match = line[/class[ \t]+[a-zA-Z0-9_]*(?=[ \t]*:[ \t]*XCTestCase)|func[ \t]+test[a-zA-Z0-9_]*(?=[ \t]*\(\))/, 0]
+    if match
+      if match[/class/, 0] == 'class'
+        class_name = match.sub(/^class[ \t]+/, '')
+        #
+        # Create a new class / func structure
+        # and add it to the classes array.
+        #
+        current_class = [class_name, []]
+        classes << current_class
+      else # Must be a func
+        func_name = match.sub(/^func[ \t]+/, '')
+        #
+        # Add each func name the the class / func
+        # structure created above.
+        #
+        current_class[1] << func_name
+      end
+    end
+  end
+  classes
+end
+
+#
+# Main routine
+#
+#
+
+tests_directory = 'Tests'
+
+options = GetoptLong.new(['--tests-dir', GetoptLong::OPTIONAL_ARGUMENT])
+options.quiet = true
+
+begin
+  options.each do |option, value|
+    case option
+    when '--tests-dir'
+      tests_directory = value
+    end
+  end
+  rescue GetoptLong::InvalidOption
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress exceptions.](https://app.codacy.com/app/FabrizioBrancati/SwiftyBot/pullRequest?prid=3796373)